core/src/components/popover/utils.ts

```diff
@@ -837,32 +837,72 @@ export const calculateWindowAdjustment = (
   const triggerHeight = triggerCoordinates ? triggerCoordinates.height : 0;
   let addPopoverBottomClass = false;
 
+  /**
+   * Approximate the safe area margins. Getting exact values would necessitate
+   * using window.getComputedStyle(), which is very expensive, so we use "close
+   * enough" values for now.
+   *
+   * 5% is derived from the iPhone 14 top safe area margin (47pt / 844 pt ~= 0.05).
+   * Source: https://useyourloaf.com/blog/iphone-14-screen-sizes/
+   *
+   * TODO(FW-5982): Investigate a more robust solution that uses the actual
+   * safe area margins through alternate means.
+   */
+  let horizontalSafeAreaApprox = 0;
+  let verticalSafeAreaApprox = 0;
+  if (win?.matchMedia !== undefined) {
+    verticalSafeAreaApprox = win.innerHeight * 0.05;
+
+    /**
+     * We only want to check horizontal safe area on landscape.
+     * Most devices do not have horizontal safe area margins in
+     * portrait mode, so enforcing it would lead to popovers
+     * being misaligned with the trigger when we don't want them
+     * to move.
+     */
+    if (win.matchMedia('(orientation: landscape)').matches) {
+      horizontalSafeAreaApprox = win.innerWidth * 0.05;
+    }
```

Comment:
I'm not really sure about this. I wanted to cover phones rotated into landscape mode, but it also causes the horizontal safe area to kick in on desktop, and possibly tablets when we don't want it to. I just couldn't think of a better way to cover things that doesn't make too many assumptions about more specific screen sizes.

One tweak that seemed promising is to add a max value for the margins, to avoid them being too large on wider devices, but I'm not sure if that's a route we'd want to take or what the max value should be.

Coming back to this PR after a while, I would be open to closing it until we can investigate a more robust solution, possibly re-opening it later if we don't determine the in-progress solution to be solidly feasible.

Reply:
I think this approach makes sense. The top/bottom safe areas always apply, but the left/right areas generally only apply for landscape. This approach is trying to get us 90% of the way there, so I think it's fair that this may not be perfectly accurate in all scenarios.

Reply:
Met with the team; we agreed to proceed with this approach, but add an undocumented escape hatch that devs can use to revert to the old behavior if they run into problems.